app/controllers/budgets_controller.rb

```diff
@@ -0,0 +1,47 @@
+class BudgetsController < ApplicationController
+  before_action :require_login
+
+  def index
+    @budgets = Budget.all
+  end
+
+  def new
+    @budget = Budget.new
+  end
+
+  def create
+    @budget = Budget.create(budget_params)
```

Comment:
You can rewrite this to be:
```
@budget = current_user.budgets.create(budget_params)
```
And thus remove line 14.

In the current way, you are making two calls to the database, one to create a budget, and one to update it with `user_id`. By doing it the way I suggest, it will be my database query to create a budget that is associated with that user.

Reply:
Love this! Done sha: 7807e6f